cluster_experiments/power_analysis.py

```diff
@@ -524,3 +524,43 @@ def check_inputs(self):
         self.check_target_col()
         self.check_treatment()
         self.check_clusters()
+
+
+class PowerAnalysisWithPreExperimentData(PowerAnalysis):
```

Comment:
I'd also override check_inputs, by checking that  isinstance(self.analysis, SyntheticControlAnalysis). or that hasattr(self.analysis, _split_pre_experiment_df) or something like this.